components/director/internal/authenticator/middleware.go

```diff
@@ -120,8 +125,9 @@ func (a *Authenticator) getBearerToken(r *http.Request) (string, error) {
 
 func (a *Authenticator) contextWithClaims(ctx context.Context, claims Claims) context.Context {
 	ctxWithTenant := tenant.SaveToContext(ctx, claims.Tenant)
```

Comment:
What about refactoring this in a way to handle tenant IDs in a struct and storing this struct in the context?